src/main/java/org/folio/dcb/service/impl/LendingLibraryServiceImpl.java

```diff
@@ -62,4 +64,15 @@ private void saveDcbTransaction(String dcbTransactionId, DcbTransaction dcbTrans
     transactionRepository.save(transactionEntity);
   }
 
+  @Override
+  public void updateTransactionStatus(TransactionEntity dcbTransaction, TransactionStatus transactionStatus) {
+    log.info("updateTransactionStatus:: updating dcbTransaction {} to status {} ", dcbTransaction, transactionStatus);
+    if (TransactionStatus.StatusEnum.OPEN.equals(dcbTransaction.getStatus())) {
+      dcbTransaction.setStatus(transactionStatus.getStatus());
```

Comment:
Dont you want to check the status before updating it ? Like check if the current status of the transaction is Created ?

Reply:
Added status check before updating